### PR TITLE
Barcode scanner setup

### PR DESCRIPTION
**Pull request by BikramDroid**

### CODE REVIEW

File: app/src/main/java/com/bikram/onboardingapp/data/ScannerRepository.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                    }
+                }.addOnFailureListener {
+                    it.printStackTrace()
+                }
+            awaitClose { }
+        }
+    }
+
+    private fun getDetails(barcode: Barcode): String {
+        return if (barcode.displayValue!!.contains("SUPERMARKET-"))
```
Comment (arunvijayanpillai) [inline app/src/main/java/com/bikram/onboardingapp/data/ScannerRepository.kt:39]: Do not use the !! as much as possible Try using the ? and handle the edge case as well.

File: app/src/main/java/com/bikram/onboardingapp/di/AppModule.kt [hunk 2]
```diff
@@ -38,4 +49,42 @@ object AppModule {
...
+        return GmsBarcodeScannerOptions.Builder()
+            .setBarcodeFormats(Barcode.FORMAT_ALL_FORMATS)
+            .build()
+    }
+
+    @ViewModelScoped
+    @Provides
+    fun providesBarCodeScanner(
+        context: Context,
+        options: GmsBarcodeScannerOptions
```
Comment (arunvijayanpillai) [inline app/src/main/java/com/bikram/onboardingapp/di/AppModule.kt:76]: Nice that you separated the options and actual scanner

File: app/src/main/java/com/bikram/onboardingapp/ui/screens/MasterPage.kt [hunk 3]
```diff
@@ -77,12 +78,32 @@ private fun MasterPageRegular(
     val appBarTitle = remember { mutableStateOf("") }
 
     val homeViewModel: HomeViewModel = hiltViewModel()
+    val barcodeState by homeViewModel.scannerUiState.collectAsState()
+
+    val barcode = barcodeState.barcodeDetails
+    if (barcode.isNotEmpty()) {
+        barcodeState.barcodeDetails = ""
```
Comment (arunvijayanpillai) [inline app/src/main/java/com/bikram/onboardingapp/ui/screens/MasterPage.kt:85]: Why change the state from composable ? cant it be done from the ViewModel ?

**Review by arunvijayanpillai — APPROVED** [review]

File: app/src/main/java/com/bikram/onboardingapp/data/ScannerRepository.kt (hunk 1 above)
Comment (BikramDroid) [reply app/src/main/java/com/bikram/onboardingapp/data/ScannerRepository.kt:39]: yes, this block was more for demonstration purpose so later when we get more production/technicaø details of barcode, then we can modify this as per requirements, but yes will handle null case and use safe nav operator

File: app/src/main/java/com/bikram/onboardingapp/ui/screens/MasterPage.kt (hunk 3 above)
Comment (BikramDroid) [reply app/src/main/java/com/bikram/onboardingapp/ui/screens/MasterPage.kt:85]: good find : )

**Comment by BikramDroid** [comment]
merging the PR, will fix below two comments in the next PR for manual search:

- safe nav operator in scanning repo
- clear barcode details from VM

**Pull request merged**